### DISCUSSION TITLE
How to access tokens?

### DISCUSSION THREAD

**Discussion by ChristianMurphy** [micromark/micromark#12] [Q&A] [answered]
The readme mentions

> to parse the entirety of markdown into tokens

but as of https://github.com/micromark/micromark/blob/66f33dd1d3c22f7ed8ff0d12d38dd8cc5cba4165/readme.md doesn't mention how to access tokens.

Is there a way to access to tokens? If so, how?

**Comment by wooorm** [comment, marked as answer]
That’s indeed on the todo. There are lower-level APIs currently, to access that (for now, see [code for core](https://github.com/micromark/micromark/blob/main/lib/index.js)). In the future this ideally would be documented, although I’m wondering who’ll be interested in it all.

The types of tokens are [here](https://github.com/micromark/micromark/blob/main/lib/constant/types.js).
